server/planning/events/events_files.py

```diff
@@ -59,13 +62,15 @@ def on_create(self, docs):
                         "length": _file.length,
                     }
 
-    def on_created(self, docs):
+    async def on_created_async(self, docs):
         for doc in docs:
             # check if the filename contains a folder, if so just return the file name component
             if isinstance(doc.get("media"), dict) and "/" in doc.get("media", {}).get("name", ""):
                 doc["media"]["name"] = doc["media"]["name"].split("/")[1]
 
-    def on_delete(self, doc):
-        events_using_file = get_resource_service("events").find(where={"files": doc.get("_id")})
-        if events_using_file.count() > 0:
+    async def on_delete_async(self, doc):
+        events_service = EventResourceModel.get_service()
+        search_request = SearchRequest(where={"files": doc.get("_id")})
+        events_using_file = await events_service.find(search_request)
+        if await events_using_file.count() > 0:
```

Comment:
You can use the `if await events_service.count({"files": doc.get("_id")}) > 0:` instead here instead, which does what you're doing here but in 1 step